### DISCUSSION TITLE
🤺 Milestones, labels and feature prioritization

### DISCUSSION THREAD

**Discussion by pranavrajs** [chatwoot/chatwoot#5803] [General]
This guide is written to outline how issues are triaged and picked up in upcoming sprints. This guide is written to make sure that the community is aware of various product boards and the changes we are bringing in for future product iterations and that the decisions are publicly available. You can see different product boards linked below and how we use them internally to track and prioritize the issues, and feature requests.

**Note:**

- The source of truth of this document would reside in the link [here](https://chatwoot.help/hc/handbook/en/processes/284). If any changes are done as a part of the discussion here, that should be updated in the link as well.
- Comments unrelated to this discussion on the thread would be deleted without request.

## The tool we use:

We used [Github Projects] previously.

As of now, we use [Zenhub](https://app.zenhub.com/workspaces/product-board-6369795c88e3ba00104e9bae/board) to track and manage issues in Chatwoot.

#### Quick Links:

- [Full Product Board](https://app.zenhub.com/workspaces/product-board-6369795c88e3ba00104e9bae/board)
- [Customer reported issues ](https://app.zenhub.com/workspaces/customer-reported-issues-63696b6204e20400103d71ac/board)
- [Untriaged issues](https://github.com/chatwoot/chatwoot#workspaces/chatwoot-product-board-6369795c88e3ba00104e9bae/board?labels%3Anot=need-more-info,need-spec,ready-for-pickup,investigation,need-discussion)
- [Under-investigation issues](https://github.com/chatwoot/chatwoot#workspaces/under-investigation-63753f4a5ce5c515803c7777/board)
- [Waiting for design input](https://app.zenhub.com/workspaces/design-board-63753f03de39670022194480/board)
- [Ready for pickup](https://app.zenhub.com/workspaces/engineering-backlog-60c6f9ede5ab53001596a073/board)

## Properties of a GitHub Issue:

- **Title & Description:** Each issue should have a proper title and detailed instructions explaining the problem (or context about the feature request)
- **Reported Date:** Every issue added to Github would have a reported date, this can be the date when the issue was created or even before that in case of live chat or email support queries.
- **Estimated Points:** Once the issue is triaged, points should be added to the ticket. (How to estimate an issue guide is available [here](https://chatwoot.help/hc/handbook/en/processes/8))
- **Release version (Milestone):** Based on the requirements, every ticket should have an associated milestone and should be prioritized in a sprint.

## General guidelines:

- All issues [should be triaged and estimated](https://chatwoot.help/hc/handbook/en/processes/8) before picking them up.
- Each issue should have a clear description and a to-do and it should follow the issue templates.
- All the 8 and 5-pointer tickets should be broken down into 2 or 3-pointer tickets.

## Triaging an issue:

The goal is to triage an issue as soon as possible and to make sure that there is a consistent response from the Chatwoot team to the customers and to the community.

![image](https://user-images.githubusercontent.com/2246121/200225030-f6fa4f5b-2a6a-4c3d-a49e-73d5d699c508.png)

### Triaging a bug:

Every issue would be considered untriaged unless it has a `ready-for-pickup` or `need-more-info` or `need-discussion` or `investigation` label.

- If the issue is reported as a bug, the assignee should try to reproduce the issue and make sure that the case is valid or mark it as **need-more-info** by requesting the information from the person who created the issue.
- If the bug is confirmed then it should have a severity label as defined [here](https://chatwoot.help/hc/handbook/en/processes/8).
- If the issue requires more time to be investigated, the assignee can mark it as an **investigation** ticket. We should make sure that the issues are not in the investigation queue for a long time
- If the request is a feature request and you cannot immediately respond, add the labels **need-discussion** and **feature-request**.

### Working on new features:

Every feature request needs a product spec. This is done to make sure that we have collected enough information to move ahead and that there is no confusion around the feature when we are trying to build one.
- Whenever you need input from the design team, add the label **need-design**.
- Once the design is complete and the product spec is complete, there can a discussion on how this would be split into multiple smaller engineering issues. (These discussions should be done async, create a meeting only if absolutely required)
- Once smaller engineering issues are created, this can be prioritized for pick up in the upcoming sprints.

## Weekly progress retro:

Every week the team would make sure that we complete the following tasks.

- Look at the previous week's issues, and see if we have missed anything.
- Follow the template of What went wrong / What can be improved / What went well to discuss the previous week's progress.
- Look at the list of customer-reported issues on the call. Prioritize the issues based on the requirements.
- Check where do we stand on the milestone? (Look at tagged issues for the milestone, and see if we need to move something)

#### Goals:

- Every issue should be triaged within one business day.
- [Severity 1 & 2 ](https://chatwoot.help/hc/handbook/en/processes/8)bugs should be fixed within a week.

#### What should be done:

When a customer (cloud or paid) reports an issue:
- Create the issue in either the Chatwoot main repo or the product repo depending on the information that will be shared. If it has sensitive information,  then create it in the [product](https://github.com/chatwoot/product/) repo.
- Tag the issue with **customer-reported-issues** on the label.
- Add it to the project Engineering Board.

In most cases, this would be done by the people who are talking to the customers directly.

#### What is expected:

- Triage the issue. Can you confirm whether the issues exist or not?
- Assess the severity of the bug if it exists. Assign the [severity](https://chatwoot.help/hc/handbook/en/processes/8) label to the ticket.

#### Where to track:
You can view all the issues here: [Link to the Engineering Board - Customer Reported Issues](https://github.com/orgs/chatwoot/projects/19/views/29). Every team member has to check the board every working day so that there are no issues unattended for a longer period of time.

#### Goals:

- Every issue should be triaged within four business days.
- [Severity 1 & 2 ](https://chatwoot.help/hc/handbook/en/processes/8)bugs should be fixed within the next release.

#### What is expected:

- A team member should be assigned to the ticket automatically in a round-robin fashion.
- Triage the issue. Can you confirm whether the issues exist or not?
- Assess the severity of the bug if it exists. Assign the [severity](https://chatwoot.help/hc/handbook/en/processes/8) label to the ticket.

## What has changed?

- 2 weeks sprint would be moved to 1-week sprints until the goals are met. Once the goals are consistently met, we would reassess this change.
- The label #enhancement is removed. Everything is a feature request or a bug. Whether it is a large feature or enhancement should be assessed using the estimated points.
- Every repo should have all the labels defined in the document [here](https://chatwoot.help/hc/handbook/en/processes/8). 

## Automations

- Every issue would be automatically assigned to a person. The project and reported date would also be added.
- The issues with `need-more-info` label would be closed after 14 days of inactivity.

**Comment by flashjames** [comment]
Why cant we login to see Zenhub boards anymore?

I just started to use chatwoot, but it would be nice to see the roadmap

**Comment by luizeof** [comment]
hello @pranavrajs the milestones are not fed so that we can follow the progress of the development. will it be removed?
